File: Freelances/Hangman/Stack.java
public class Stack<T> {
    private int maxSize;
    private T[] stackArray;
    private int top;

    public Stack(int maxSize) {
        this.maxSize = maxSize;
        this.stackArray = (T[]) new Object[maxSize];
        this.top = -1;
    }

    public void push(T value) {
        if (isFull()) {
            //System.out.println("Stack is full. Cannot push element.");
            return;
        }
        stackArray[++top] = value;
    }

    public T pop() {
        if (isEmpty()) {
            //System.out.println("Stack is empty. Cannot pop element.");
            return null;
        }
        return stackArray[top--];
    }

    public T peek() {
        if (isEmpty()) {
            //System.out.println("Stack is empty. Cannot peek element.");
            return null;
        }
        return stackArray[top];
    }

    public boolean isFull() {
        return (top == maxSize - 1);
    }

    public boolean isEmpty() {
        return (top == -1);
    }

    public int size() {
        return top + 1;
    }

    public T get(int id)
    {
        return  stackArray[id];
    }
    public void set(T obj , int id)
    {
        stackArray[id] = obj;
    }

    public boolean contains(T value) {
        for (int i = 0; i <= top; i++) {
            if (stackArray[i].equals(value)) {
                return true;
            }
        }
        return false;
    }

}

File: Freelances/Unesco Miras/UnescoMiras/UnescoMiras/Stack.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace UM
{
    public class StackX<T>
    {
        private int maxSize; // size of stack array
        private T[] stackArray;
        private int top; // top of stack

        //--------------------------------------------------------------
        public StackX(int s = 20) // constructor
        {
            maxSize = s; // set array size
            stackArray = new T[maxSize]; // create array
            top = -1; // no items yet
        }

        //--------------------------------------------------------------
        public void Push(T item) // put item on top of stack
        {
            if (top == maxSize - 1)
            {
                throw new InvalidOperationException("Stack is full");
            }

            stackArray[++top] = item; // increment top, insert item
        }

        //--------------------------------------------------------------
        public T Pop() // take item from top of stack
        {
            if (top == -1)
            {
                throw new InvalidOperationException("Stack is empty");
            }

            return stackArray[top--]; // access item, decrement top
        }

        //--------------------------------------------------------------
        public T Peek() // peek at top of stack
        {
            if (top == -1)
            {
                throw new InvalidOperationException("Stack is empty");
            }

            return stackArray[top];
        }

        //--------------------------------------------------------------
        public bool IsEmpty() // true if stack is empty
        {
            return (top == -1);
        }

        //--------------------------------------------------------------
        public bool IsFull() // true if stack is full
        {
            return (top == maxSize - 1);
        }
        //--------------------------------------------------------------

        public T this[int index]
        {
            get
            {
                if (index < 0 || index > top)
                {
                    throw new IndexOutOfRangeException("Index is out of range");
                }

                return stackArray[index];
            }
            set
            {
                if (index < 0 || index > top)
                {
                    throw new IndexOutOfRangeException("Index is out of range");
                }

                stackArray[index] = value;
            }
        }

    }
    public class QueueX<T>
    {
        private int maxSize; // size of queue array
        private T[] queueArray;
        private int front; // front of the queue
        private int rear; // rear of the queue
        private int itemCount; // current number of items in the queue

        //--------------------------------------------------------------
        public QueueX(int s = 20) // constructor
        {
            maxSize = s; // set array size
            queueArray = new T[maxSize]; // create array
            front = 0; // initialize front
            rear = -1; // initialize rear
            itemCount = 0; // no items yet
        }

        //--------------------------------------------------------------
        public void Enqueue(T item) // insert item at rear of queue
        {
            if (itemCount == maxSize)
            {
                throw new InvalidOperationException("Queue is full");
            }

            if (rear == maxSize - 1)
            {
                rear = -1; // wrap around
            }

            queueArray[++rear] = item; // increment rear, insert item
            itemCount++;
        }

        //--------------------------------------------------------------
        public T Dequeue() // remove item from front of queue
        {
            if (itemCount == 0)
            {
                throw new InvalidOperationException("Queue is empty");
            }

            T temp = queueArray[front++]; // get value and increment front
            if (front == maxSize)
            {
                front = 0; // wrap around
            }

            itemCount--;
            return temp;
        }

        //--------------------------------------------------------------
        public T Peek() // peek at front of queue
        {
            if (itemCount == 0)
            {
                throw new InvalidOperationException("Queue is empty");
            }

            return queueArray[front];
        }

        //--------------------------------------------------------------
        public bool IsEmpty() // true if queue is empty
        {
            return (itemCount == 0);
        }

        //--------------------------------------------------------------
        public bool IsFull() // true if queue is full
        {
            return (itemCount == maxSize);
        }

        //--------------------------------------------------------------
        public int Count() // number of items in the queue
        {
            return itemCount;
        }
    }
}

